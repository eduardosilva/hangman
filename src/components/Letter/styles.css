@import '../../styles/variables.css';

.letterWrapper {
  background-color: rgba(83, 85, 93, 0.95);
  border-radius: 5px;
  color: var(--letterColor);
  display: inline-block;
  height: 70px;
  letter-spacing: 0;
  line-height: 74px;
  text-align: center;
  margin: 5.6px 5.6px 5.6px 0;
  vertical-align: text-bottom;
  width: 62px;
}
.letterWrapper:last-child {
  margin-right: 0;
}
