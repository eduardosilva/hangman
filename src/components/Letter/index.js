import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import styles from './styles.css'

export const Letter = ({ letter, masked }) => (
  <span className={cx(styles.letterWrapper, { blank: masked })}>{!masked && letter}</span>
)

Letter.propTypes = {
  letter: PropTypes.string.isRequired,
  masked: PropTypes.bool.isRequired,
}

export default Letter
