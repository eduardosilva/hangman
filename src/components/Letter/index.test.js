import React from 'react'
import renderer from 'react-test-renderer'
import Letter from '../Letter'

it('should render a letter', () => {
  const letter = renderer
    .create(<Letter letter="a" masked={false} />)
    .toJSON()
  expect(letter).toMatchSnapshot()
})
