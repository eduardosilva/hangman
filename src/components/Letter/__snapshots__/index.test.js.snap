// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a letter 1`] = `
<span
  className=""
>
  a
</span>
`;
